src/helics/core/mpi/MpiComms.h

```diff
@@ -10,29 +10,96 @@ This software was co-developed by Pacific Northwest National Laboratory, operate
 #define _HELICS_MPI_COMMS_
 #pragma once
 
-#include "helics/core/CommsInterface.h"
+#include "../CommsInterface.hpp"
 #include <atomic>
+#include <set>
+#include <string>
+#include <future>
+#include <vector>
+#include <mutex>
+#include "helics/helics-config.h"
 
+#include <mpi.h>
+
+#if (BOOST_VERSION_LEVEL >=2)
+namespace boost
+{
+    namespace asio
+    {
+        class io_context;
+        using io_service = io_context;
+    }
+}
+#else
+namespace boost
+{
+    namespace asio
+    {
+        class io_service;
+    }
+}
+#endif
 namespace helics {
 
-/** implementation for the core that uses zmq messages to communicate*/
-class MpiComms:public CommsInterface {
+/** generate a string with a full address based on an interface string and port number
+@details,  how things get merged depend on what interface is used some use port number some do not
+
+@param[in] interface a string with an interface description i.e 127.0.0.1 
+@param portNumber the number of the port to use
+@return a string with the merged address
+*/
+std::string makePortAddress(const std::string &networkInterface, int portNumber);
+
+/** extract a port number and interface string from an address number
+@details,  if there is no port number it default to -1 this is true if none was listed
+or the interface doesn't use port numbers
+
+@param[in] address a string with an network location description i.e 127.0.0.1:34
+@return a pair with a string and int with the interface name and port number
+*/
+std::pair<std::string, int> extractInterfaceandPort(const std::string &address);
```

Comment:
Do you actually use these functions in MPI comms?  if so I would recommend using the ones in NetworkBrokerData and get rid of these definitions since they will be duplicated